src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +101,659 @@ const ActorInit En_Nb_InitVars = {
+                                          &this->actor);
+                    this->unk_288++;
+                    ret = true;
+            }
+            break;
+
+        case 1:
+        case 3:
+        case 5:
+        case 7:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)),
+                                      this->actor.child);
+            }
+            this->unk_288++;
+            ret = true;
+            break;
+
+        case 9:
+            ActorCutscene_Stop(cutscene);
+            this->unk_288++;
+            ret = true;
+            break;
+    }
+
+    return ret;
+}
+
+s32 func_80BC01DC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s32 pad;
+    s32 ret = false;
+
+    switch (this->unk_288) {
+        case 0:
+            if (Player_GetMask(play) == PLAYER_MASK_ALL_NIGHT) {
+                this->unk_288 = 1;
+            } else {
+                this->unk_288 = 5;
+            }
+            break;
+
+        case 1:
+            func_8016A268(&play->state, 1, 0, 0, 0, 0);
+            this->unk_286 = 40;
+            this->unk_288 = (u16)(this->unk_288 + 1);
+            break;
+
+        case 2:
+            MREG(68) = (s16)(s32)(255.0f - (((f32)ABS_ALT(20 - this->unk_286) / 20.0f) * 255.0f));
+
+            if (this->unk_286 == 20) {
+                if (gSaveContext.eventInf[4] & 4) {
+                    play->interfaceCtx.unk_31B = 0;
+                } else {
+                    play->interfaceCtx.unk_31B = 1;
+                }
+                play->interfaceCtx.unk_31A = 6;
+                XREG(91) = 0xFF;
+            }
+
+            if (DECR(this->unk_286) == 0) {
+                this->unk_288++;
+            }
+            break;
+
+        case 3:
+            play->interfaceCtx.unk_31A = 4;
+            this->unk_288++;
+            ret = true;
+            break;
+
+        case 4:
+            play->interfaceCtx.unk_31A = 5;
+            this->unk_288++;
+            // fallthrough
+        case 5:
+            if (!(gSaveContext.eventInf[4] & 4)) {
+                gSaveContext.save.time = CLOCK_TIME(8, 0);
+                Sram_IncrementDay();
+            } else {
+                func_800FE658(120.0f);
+            }
+
+            this->unk_288++;
+            play->nextEntrance = ENTRANCE(STOCK_POT_INN, 2);
+            gSaveContext.nextCutsceneIndex = 0;
+            play->transitionTrigger = TRANS_TRIGGER_START;
+            play->transitionType = TRANS_TYPE_02;
+            gSaveContext.nextTransitionType = TRANS_TYPE_06;
+            gSaveContext.eventInf[4] |= 8;
+            break;
+    }
+
+    return ret;
+}
+
+UNK_PTR func_80BC045C(EnNb* this, PlayState* play) {
+    if (gSaveContext.eventInf[4] & 8) {
+        this->unk_28C = func_80BC01DC;
+        return &D_80BC1464;
```

Comment:
```suggestion
        return D_80BC1464;
```
Since they are arrays, they shouldn't need `&`s.